base/hashing.jl

```diff
@@ -132,7 +186,12 @@ function hash(x::Real, h::UInt)
         den = -den
     end
     num_z = trailing_zeros(num)
-    num >>= num_z
+
+    if x isa BigFloat
```

Comment:
is it better to just make a new method? would be a lot of duplicated code, but `isa` in control flow always feels weird to me